enhancements/olm-toggle-copied-csvs.md

```diff
@@ -0,0 +1,163 @@
+
+- there exists at least one`Operator` resource for every CSV
+- there exists at least one `Operator` resource for every `Subscription`
+
+That is, all CSVs installed by a given `Subscription` share an `Operator` resource __and__ any CSVs that weren't installed by a `Subscription` are related to their own `Operator` resource.
+
+#### Orphaned CSVs
+
+Orphaned CSVs are CSVs that were installed via a `Subscription` that no longer exists. Since the lifetimes -- i.e. the period of existence on a cluster --  of `Operator` resources aren't directly tied to the lifetimes of the resources that triggered their generation, the `Operator` resource(s) associated with Orphaned CSVs will continue to exist _after_ a `Subscription` is deleted. If that `Operator` resource is deleted explicitly, OLM will generate another as per the rules [outlined above](#operator-cardinality).
+
+#### Copied RBAC 
+
+In addition to operator discoverability, Copied CSVs also serve as convienient vehicles for the garbage collection of Copied RBAC. These are resources that are copied along with the CSV to provide the operator access to a namespace. To support the loss of Copied CSVs, the RBAC copying logic in OLM will be extended to write additional owner references from copied resources to the related Operator resource. 
+
+#### The Toggle
+
+A feature toggle will be added to control OLM's use of Copied CSVs. When enabled (toggled on), OLM will delete all existing Copied CSVs and will not generate any more. When disabled (toggled off), OLM will generate Copied CSVs (much like it does today).
+
+When toggled (on then off), OLM will recreate any missing Copied CSVs and reassociate the owner references of related copied resources with them (i.e. the inverse operation of the one described in the [Copied RBAC](#copied-rbac) section).
+
+The toggle will be disabled by default.
+
+The toggle can be set from either a command-line option on the olm-operator binary _or_ a field of the `olm-config` `ConfigMap` colocated in the same namespace as OLM's deployment on-cluster; when both are present, the field of the `ConfigMap` takes precedence.
+
+### Risks and Mitigations
+
+#### Loss of Operator Discovery
+
+**Risk:** When Copied CSVs are disabled, users won't be able to discover what operators are configured to watch a particular namespace unless they have access to their install namespaces.
+**Potential Mitigations:** 
+
+- Console displays a warning to users when it detects that Copied CSVs are disabled
+- OLM fires an alert in all affected namespaces on startup when Copied CSVs are disabled
+- Docs suggest admins notify their cluster tenants of installed operators directly; i.e. email, slack, etc
+
+#### OpenShift Console Integration
+
+**Risk:** The OpenShift Console uses Copied CSVs in the rendering of certain pages.
```

Comment:
AFAIK other than the "Installed Operators" piece, the console (both admin and dev perspectives) relies heavily on Copied CSVs for rendering "Operands-related" pages. 

The info arch in the admin console is:
```
"Installed Operators" page (i.e. a list of Copied CSVs) 
└── "Operator Details" page (e.g. Copied CSV of Strimzi)
            └── "Operand List" page (i.e. instance list of provided APIs through Copied CSV)
                        └── "Operand Details" page (e.g. kafka instance details through Copied CSV)
```
 (For dev console: end-users directly create Operand instance from Dev Catalog page through Copied CSV)

So the risk of Copied CSVs being removed/toggled off is that the end-users (who only have access to their namespaces) would lose the ability to create/manage/delete Operand instances in the console. 

**_e.g._** if Copied CSV of Strimzi Operator is gone, as regular users who only have access to “dev” namespace, they can’t create kafka instance in the console through:
`/k8s/ns/dev/clusterserviceversions/strimzi-cluster-operator.v0.26.0/kafka.strimzi.io~v1beta2~Kafka/~new`


I think it's worth calling out the UI impact when toggled off Copied CSV in the doc:
- if toggle off the Copied CSV, basically all UI workflows for end-users to manage Operand instances are crippled, so cluster admins are well-aware of this side-effect

- (I assume this "Copied CSV toggle" is meant for a short-term way to reduce the cluster-loading) so it's important to let cluster admins know they can toggle Copied CSV back on to re-enable Operand instance managing workflow in the console for end-users.

Reply:
@tlwu2013 Thanks for this!

I used your comment to flesh out the Risks/Mitigations. Now, between this section and the "Loss of Operator Discovery" section before it, I think I've captured your concerns. PTAL.